### PR TITLE
sphinx: switch to `openssl@3`

### PR DESCRIPTION
**Pull request by mchoo7**
See https://github.com/Homebrew/homebrew-core/issues/134251.

### CODE REVIEW

**Comment by fel1x-developer** [comment]
[sphinx github repo](https://github.com/sphinxsearch/sphinx)

> This is the 2.x branch, not really supported or actively developed as of 2018.

The source code of 3.x is not released yet.
I think 2.x doesn't support `openssl@3`.

**Comment by chenrui333** [comment]
- needs #134421

no openssl linkage on linux build

**Comment by fel1x-developer** [comment]
> * needs [mysql@5.7: switch to `openssl@3` #134421](https://github.com/Homebrew/homebrew-core/pull/134421)
> 
> no openssl linkage on linux build

Thanks!

File: Formula/sphinx.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class Sphinx < Formula
   url "https://sphinxsearch.com/files/sphinx-2.2.11-release.tar.gz"
   sha256 "6662039f093314f896950519fa781bc87610f926f64b3d349229002f06ac41a9"
   license "GPL-2.0-or-later"
-  revision 4
+  revision 5
```
Comment (chenrui333) [inline Formula/sphinx.rb:7]: no need to specify revision then

**Comment by chenrui333** [comment]
it does have openssl linkage for macos builds, see run, https://github.com/Homebrew/homebrew-core/actions/runs/5361110431/jobs/9726804943

**Comment by fel1x-developer** [comment]
> it does have openssl linkage for macos builds, see run, https://github.com/Homebrew/homebrew-core/actions/runs/5361110431/jobs/9726804943

https://github.com/Homebrew/homebrew-core/pull/134421
`mysql@5.7` won't support `openssl@3`. I think `sphinx` will be deprecated when `mysql@5.7` deprecates in October. Even if we add support for `openssl@3` here, it will conflict with `openssl@1.1` for `mysql@5.7`.

**Comment by carlocab** [comment]
Closing since we can't migrate `mysql@5.7`. Thanks anyway @fel1x-developer!

**Pull request closed without merging**